src/commands/moderators/deleteAllMessage.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  ApplicationCommandType,
+  ChannelType,
+  DiscordAPIError,
+  GuildMember,
+  PermissionsBitField,
+} from 'discord.js'
+
+import { CommandHandlerConfig } from '../../types/CommandHandlerConfig.js'
+
+export default {
+  data: {
+    name: 'Delete All',
+    type: ApplicationCommandType.Message,
+  },
+  ephemeral: true,
+  execute: async (client, interaction) => {
+    if (!interaction.guild || !interaction.isContextMenuCommand()) return
+
+    // Get the member who initiated the interaction
+    const member = interaction.member as GuildMember
+
+    // Check user permission
+    if (!member.permissions.has(PermissionsBitField.Flags.ManageChannels)) {
+      console.warn(`[WARN] ${member.displayName} try to delete all message`)
+      await interaction.editReply(
+        'Invalid permission, this command available for ONLY Moderator',
+      )
+      return
+    }
+
+    // Delete message in all channel
+    client.channels.cache.forEach(async (channel) => {
```

Comment:
Using forEach with async causes the callback function to run in parallel without a way to wait.

Recommendation:

- Option 1 (preferred): Use `for .. of` to process each channel one-by-one.
- Option 2: Use `Promise.all()` or `p-map` to await the processing of each channel.